### ISSUE TITLE
Discrepancy between MyPy and Pyright library module resolution

### ISSUE THREAD

**Issue by NeilGirdhar** [microsoft/pyright#5858]
```python
import jax.numpy as jnp
reveal_type(jnp.linalg)
```
Gives:
```
❯ mypy a.py
a.py:2: note: Revealed type is "types.ModuleType"
Success: no issues found in 1 source file
❯ pyright a.py
/home/neil/src/tjax/a.py
  /home/neil/src/tjax/a.py:2:17 - error: "linalg" is not a known member of module "jax" (reportGeneralTypeIssues)
  /home/neil/src/tjax/a.py:2:13 - information: Type of "jnp.linalg" is "Unknown"
```
The code in question has `jax.numpy.__init__.py`:
```
from jax.numpy import linalg as linalg
```
where `jax.numpy.linalg.__init__.py` is a module.

Should PyRight find this, or is there something Jax can do to make it more obvious.  None of the `__init__.py` files have an `__all__`.

**Comment by erictraut** [comment]
I'm not able to repro what you're seeing. Here's the output I receive when I run pyright 1.1.325 on your code sample:

```
test.py
  test.py:3:13 - information: Type of "jnp.linalg" is "Module("jax.numpy.linalg")"
```

I checked that I have the latest version of `jax` installed. Perhaps you're using an older version?

**Comment by NeilGirdhar** [comment]
Interesting, Poetry isn't updating Jax to 0.4.15, but rather "0.4.15.dev20230830".  Sorry, this might have something to do with either how Jax is releasing or how Poetry is resolving versions.

**Comment by NeilGirdhar** [comment]
Probably related to this: https://github.com/google/jax/issues/17371
